hive/src/test/java/com/netflix/iceberg/hive/HiveTablesTest.java

```diff
@@ -87,4 +94,38 @@ public void testFailure() throws TException {
             .addColumn("data", Types.LongType.get())
             .commit();
   }
+
+  @Test
+  public void testConcurrentFastAppends() {
+    HiveTables hiveTables = new HiveTables(hiveConf);
+    com.netflix.iceberg.Table icebergTable = hiveTables.load(DB_NAME, TABLE_NAME);
+    com.netflix.iceberg.Table anotherIcebergTable = hiveTables.load(DB_NAME, TABLE_NAME);
+
+    String fileName = UUID.randomUUID().toString();
+    DataFile file = DataFiles.builder(icebergTable.spec())
+      .withPath(FileFormat.PARQUET.addExtension(fileName))
+      .withRecordCount(2)
+      .withFileSizeInBytes(0)
+      .build();
+
+    Tasks.foreach(icebergTable, anotherIcebergTable)
+      .stopOnFailure().throwFailureWhenFinished()
+      .executeWith(getWorkerPool())
+      .run(table -> {
+        for (int numCommittedFiles = 0; numCommittedFiles < 10; numCommittedFiles++) {
+          long commitStartTime = System.currentTimeMillis();
+          table.newFastAppend().appendFile(file).commit();
+          long commitEndTime = System.currentTimeMillis();
+          long commitDuration = commitEndTime - commitStartTime;
+          try {
+            TimeUnit.MILLISECONDS.sleep(200 - commitDuration);
+          } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+          }
+        }
+      });
+
+    icebergTable.refresh();
+    Assert.assertEquals(20, icebergTable.currentSnapshot().manifests().size());
```

Comment:
You probably want to test that the actual ids are different for all commits. Even without this change I think this assertion will pass. Also this won't really guarantee that the snapshotId method is concurrently executed so I wonder if it adds any value.

Reply:
I executed this test ±25 times before the fix and it failed each time. If snapshot ids are not unique, there will be a non-recoverable runtime exception in `TableMetadata` when Iceberg builds `ImmutableMap<Long, Snapshot>`.

As we are using `ThreadPools$WORKER_POOL`, we should have 10 concurrent commit attempts. On my laptop, this was enough to reproduce the issue. Let me know if you have a better idea for testing. I will be glad to implement that.

But you are right, this check is not the most crucial part of the test. It is there to ensure we have the right number of manifests in case of concurrent writers.

Reply:
We can add a separate check to ensure that ids are unique. Just in case the logic in `TableMetadata` changes.

Reply:
Makes sense. I am ok with merging this as is given in absence of uniqueness for ids the test fails.